content/languages/lambdacalc/authoring.md

```diff
@@ -0,0 +1,120 @@
+
+Due to being an esoteric language, in many cases tasks may be significantly harder to complete in Lambda Calculus, than in other languages. Due to this, many kata translations should rather be published as standalone katas with `[LC]` as a title prefix.
+
+## Tests
+
+### lc-test
+
+Lambda Calculus kata use [lc-test](/languages/lambdacalc/lc-test/) to implement and execute tests.
+
+### Setup
+
+Lambda Calculus is heavily dependant upon what encodings are chosen to represent various data types. All Lambda Calculus kata should declare in the description which `Purity` and which `numEncoding` (if any) are going to be used in the kata. Additionally, any required encodings should also be described, such as expected input and output types.
+
+### What is in what language, what syntax, what encoding?
+
+The `Preloaded` section, example `Solution`, and solver `Solution` code are `String`s in Lambda Calculus syntax. The sample and submission tests are written and run in Javascript.
+
+`LC.compile` compiles Lambda Calculus code to JavaScript `Function`s. There is a one-to-one correspondence between Lambda Calculus syntax and JavaScript syntax, and everything that comes out of `compile` _is JavaScript,_ and is a callable `Function`. It can be applied with native JavaScript `Function`s, with compiled Lambda Calculus terms ( thus JavaScript `Function`s ), or with other JavaScript values, which will pass uncompiled. Thus, _everything that happens in testing happens in JavaScript._
+
+The only way to arrive at JavaScript values from compiled Lambda Calculus code is passing its compiled form a JavaScript value in testing; there is no way to embed JavaScript values in Lambda Calculus code - not even passing it `Number`s, which will be transparently encoded ( or generate an `EvalError`, when `numEncoding = None` ).
+
+Note that there are _no `String`s_ in Lambda Calculus syntax, though you can pass and manipulate `String`s in testing ( i.e. in JavaScript ).
+
+Also note that _all_ passed `Number`s will be transparently encoded, which will generate a `RangeError` for negative numbers in most `numEncoding`s.
+
+### Leave solvers free to use encodings
+
+`numEncoding` is designed to set a particular numeral encoding for the kata, and provide support for testing with that. Other encodings however should be as free for solvers as possible. Use constructors and deconstructors in wrappers for testing, and allow solvers to export their own (de)constructors for datatypes other than numbers.
+
+For booleans, ask for `False, True` and some `if then else` ( TBD ).
+For 2-tuples, ask for `Pair` and `fst, snd`.
+For lists, ask for `nil, cons` and `foldr`.
+For option types, ask for `none, some` and `option`.
+
+Refer to the [Encodings Reference](encoding-reference) for naming conventions.
+
+## Example test suite
+
+Below you can find an example test suite, showing how the components of `lc-test` are used to compile and test a Lambda Calculus Kata.
```

Comment:
The shown test suite should _import_ `True` and `False`, not define it itself.

Constructors and deconstuctors define the datatype. I necessary, specify the signature for the (de)constructors).